### PR TITLE
new endpoint added in mangadex like fetchrandom,fetchRecentlyAdded,fe…

### LINKED ISSUE #439: new release and popular endpoint for mangadex
### Source name

Mangadex

### Source link

https://mangadex.org

### Language

English

### Source type

Manga

### Additional info

Mangadex endpoint should also have a New Releases or Popular

### PR DESCRIPTION
**Pull request by kodify-js**
**What kind of change does this PR introduce?**
=>refactoring

**Did you add tests for your changes?**
=>No
**If relevant, did you update the documentation?**
=>yes
**Summary**
This change resolves #439 & #451 by adding fetchRecentlyAdded  fetchLatestUpdates fetchpopular fetchrandom in Mangadex endpoint

**Other information**

### CODE REVIEW

**Review by riimuru — COMMENTED** [review]
You would mind adding these new methods to docs?

File: src/providers/manga/mangadex.ts [hunk 1]
```diff
@@ -128,7 +128,170 @@ class MangaDex extends MangaParser {
       throw new Error((err as Error).message);
     }
   };
+  fetchrandom = async (
```
Comment (riimuru) [inline src/providers/manga/mangadex.ts:131]: Can you change the function name to follow the camel case

File: src/providers/manga/mangadex.ts [hunk 2]
```diff
@@ -128,7 +128,170 @@ class MangaDex extends MangaParser {
...
+
+        return results;
+      } else {
+        throw new Error(res.data.message);
+      }
+    } catch (err) {
+      throw new Error((err as Error).message);
+    }
+  };
+  fetchpopular = async (
```
Comment (riimuru) [inline src/providers/manga/mangadex.ts:252]: Change to camel case here too

**Review by riimuru — CHANGES_REQUESTED** [review]
Thank you for adding the docs!

File: src/providers/manga/mangadex.ts (hunk 1 above)
Comment (kodify-js) [reply src/providers/manga/mangadex.ts]: ok

File: src/providers/manga/mangadex.ts (hunk 2 above)
Comment (kodify-js) [reply src/providers/manga/mangadex.ts]: ok

**Review by riimuru — APPROVED** [review]

**Pull request merged**